src/JsonSchema/Validator.php

```diff
@@ -52,4 +52,17 @@ public function coerce(&$value, $schema = null, JsonPointer $path = null, $i = n
 
         $this->addErrors(array_unique($validator->getErrors(), SORT_REGULAR));
     }
+
+    /**
+     * Does everything that coerce does, but will also set values to their default, if the value is not
+     * set and a default is available in the schema. Note that the first argumen is passwd by
+     * reference, so you must pass in a variable.
+     *
+     * {@inheritDoc}
+     */
+    public function coerceDefault(&$value, $schema = null, JsonPointer $path = null, $i = null)
```

Comment:
I like that idea a lot, and yes, quite happy to make the change. Do you want it in this PR, or in its own PR (and then I can rebase this after that one is merged)?

I don't see that use-case you mention above as a problem - anyone who needs to do something like that will just assign it to a variable and then check the variable - but it does raise the question of whether it's OK to break this functionality 'in the wild'.

I do think that Validator::coerce() should stay, if only as a flag-setting alias, to prevent breaking things for anyone who is currently using it.